src/integer_mod_q/mat_zq/vector/is_vector.rs

```diff
@@ -0,0 +1,168 @@
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes all functionality to check
+//! whether a matrix represents a vector, i.e. has only one row
+//! or one column.
+//! These methods should be used to ensure that vector functions
+//! can only be called on suitably formed vector/matrices.
+
+use super::super::MatZq;
+use crate::traits::{GetNumColumns, GetNumRows};
+
+impl MatZq {
+    /// Returns `true` if the provided [`MatZq`] has only one row,
+    /// i.e. is a row vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let col_vec = MatZq::from_str("[[1],[2],[3]] mod 4").unwrap();
+    /// let row_vec = MatZq::from_str("[[1,2,3]] mod 4").unwrap();
+    ///
+    /// assert!(row_vec.is_row_vector());
+    /// assert!(!col_vec.is_row_vector());
+    /// ```
+    pub fn is_row_vector(&self) -> bool {
+        self.get_num_rows() == 1
+    }
+
+    /// Returns `true` if the provided [`MatZq`] has only one column,
+    /// i.e. is a column vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let col_vec = MatZq::from_str("[[1],[2],[3]] mod 4").unwrap();
+    /// let row_vec = MatZq::from_str("[[1,2,3]] mod 4").unwrap();
+    ///
+    /// assert!(col_vec.is_column_vector());
+    /// assert!(!row_vec.is_column_vector());
+    /// ```
+    pub fn is_column_vector(&self) -> bool {
+        self.get_num_columns() == 1
+    }
+
+    /// Returns `true` if the provided [`MatZq`] has only one column or one row,
+    /// i.e. is a vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let col_vec = MatZq::from_str("[[1],[2],[3]] mod 4").unwrap();
+    /// let row_vec = MatZq::from_str("[[1,2,3]] mod 4").unwrap();
+    ///
+    /// assert!(col_vec.is_vector());
+    /// assert!(row_vec.is_vector());
+    /// ```
+    pub fn is_vector(&self) -> bool {
+        self.is_column_vector() || self.is_row_vector()
+    }
+
+    /// Returns `true` if the provided [`MatZq`] has only one entry,
+    /// i.e. is a 1x1 matrix. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZq::from_str("[[1]] mod 2").unwrap();
+    ///
+    /// assert!(vec.has_single_entry());
+    /// ```
+    pub fn has_single_entry(&self) -> bool {
+        self.is_column_vector() && self.is_row_vector()
+    }
+}
+
+#[cfg(test)]
+mod test_is_vector {
+
+    use super::*;
+    use std::str::FromStr;
+
+    /// Check whether matrices with one row or one column
+    /// get recognized as (row or column) vectors
+    #[test]
+    fn vectors_detected() {
```

Comment:
just thought of it now, please also add a testcase, where there are two row/cloumns, but he second only consists of zeros

Reply:
Done